src/book/03-guides/01-main-codebase/03-recording-playback.mdx

```diff
@@ -0,0 +1,120 @@
+---
+section: Guides
+chapter: Main Codebase
+title: Recording and Playback
+description: Recording data and playing it back.
+slug: /guides/main/recording-playback
+---
+
+NUbots data is stored in [NUClear Binary Stream (NBS) files](/system/foundations/nuclear#nbs-files). Any [messages](https://github.com/NUbots/NUbots/tree/master/shared/message) that are passed around the NUbots system can be recorded.
+
+## Recording Data
+
+The following steps detail how to record data. An example will be given throughout involving recording sensors messages.
+
+1. Set up the role. Most of the time this involves making minor changes to an existing role.
+
+   1. The role must contain the module/s which emit the message/s you want to record.
+   2. The role must contain the module `support::logging::DataLogging`.
+
+   For example, you may want to record [Sensors](https://github.com/NUbots/NUbots/blob/master/shared/message/input/Sensors.proto) messages. You could use the existing [sensortest](https://github.com/NUbots/NUbots/blob/master/roles/sensortest.role) role and add `support::logging::DataLogging` to it.
+
+2. Set up the configuration file. This involves making changes to the `DataLogging.yaml` file. Add in the message you want to log, specifying `true`.
```

Comment:
```suggestion
2. Set up the configuration file. This involves making changes to the [`DataLogging.yaml`](https://github.com/NUbots/NUbots/tree/master/module/support/logging/DataLogging/data/config/webotsrobocup) file. Add in the message you want to log, specifying `true`.
```

Reply:
```suggestion
2. Set up the configuration file. This involves making changes to the [`DataLogging.yaml`](https://github.com/NUbots/NUbots/tree/master/module/support/logging/DataLogging/data/config) file. Add in the message you want to log, specifying `true`.
```
rather than the webotsrobocup role specific folder